live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/properties/Retry.java

```diff
@@ -0,0 +1,89 @@
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.adapters.config.properties;
+
+import java.net.MalformedURLException;
+import java.net.UnknownHostException;
+import java.util.Set;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+import jakarta.validation.constraints.PositiveOrZero;
+
+import com.fasterxml.jackson.core.io.JsonEOFException;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Getter;
+import org.apache.hc.client5.http.HttpHostConnectException;
+import org.apache.hc.core5.http.MalformedChunkCodingException;
+import org.apache.hc.core5.http.NoHttpResponseException;
+
+import org.alfresco.hxi_connector.live_ingester.domain.exception.EndpointServerErrorException;
+
+@Data
+@Getter(AccessLevel.NONE)
+public class Retry
+{
+    private static final int RETRY_ATTEMPTS_DEFAULT = 10;
+    private static final int RETRY_INITIAL_DELAY_DEFAULT = 500;
+    private static final double RETRY_DELAY_MULTIPLIER_DEFAULT = 2;
+    private static final Set<Class<? extends Throwable>> RETRY_REASONS = Set.of(
+            EndpointServerErrorException.class,
+            UnknownHostException.class,
+            MalformedURLException.class,
+            JsonEOFException.class,
+            MismatchedInputException.class,
+            HttpHostConnectException.class,
+            NoHttpResponseException.class,
+            MalformedChunkCodingException.class);
+
+    @PositiveOrZero
+    private int attempts = RETRY_ATTEMPTS_DEFAULT;
+    @PositiveOrZero
+    private int initialDelay = RETRY_INITIAL_DELAY_DEFAULT;
+    @Positive private double delayMultiplier = RETRY_DELAY_MULTIPLIER_DEFAULT;
+    @NotNull private Set<Class<? extends Throwable>> reasons = RETRY_REASONS;
+
+    public int attempts()
+    {
+        return attempts;
+    }
+
+    public int initialDelay()
+    {
+        return initialDelay;
+    }
+
+    public double delayMultiplier()
+    {
+        return delayMultiplier;
+    }
+
+    public Set<Class<? extends Throwable>> reasons()
+    {
+        return reasons;
+    }
```

Comment:
maybe we could use Accessors(fluant=true) instead of writing this manually?

Reply:
Unfortunately Spring requires setters in traditional form, so we have to declare or fluent getters or setters in this class